test/tests/treeview-1a.js

```diff
@@ -21,50 +21,47 @@ const ex = {
 
 const openAllFolders = async function (t) {
   const closedFoldersSelector = ex.treeitemSelector + '[aria-expanded="false"]';
-  let folders = await t.context.session.findElements(By.css(closedFoldersSelector));
+  let closedFolders = await t.context.session.findElements(By.css(closedFoldersSelector));
 
-  // While their are closed folders
-  while (folders.length != 0) {
+  // While there are closed folders
+  while (closedFolders.length > 0) {
 
     // Find the first visible closed folder and click it
-    for (let folder of folders) {
-      if (await folder.isDisplayed()) {
-        await folder.click();
-        break;
-      }
-    }
+    await closedFolders[0].click();
 
-    folders = await t.context.session.findElements(By.css(closedFoldersSelector));
+    // Get the updated list of closed folders (as opening one folder may have revealled
+    // more folders)
+    closedFolders = await t.context.session.findElements(By.css(closedFoldersSelector));
```

Comment:
You should be able to do this:

```
const childClosedFolders = await folder.findElements(By.css(closedFoldersSelector));
closedFolders = closedFolders.concat(childClosedFolders);
```

Reply:
This way, you're just finding new closed folders inside the one you just opened, rather than gathering the entire list of closed folders on the page in each iteration and replacing the existing list.

Reply:
Actually, it's a bit foolish -- but I don't need to and can't appended, the selector used to find closed folders finds all closed folder dom elements regardless of whether they are visible. Which makes me realize.. the whole thing should be a lot simpler :/